avc/avc.go

```diff
@@ -0,0 +1,151 @@
+package avc
```

Comment:
I think we should move this package into "internal/avc" and not export it outside the repo for now. If we export it, we should improve the documentation.